src/metabase/models/params.clj

```diff
@@ -102,19 +101,18 @@
   (get-in card [:dataset_query :native :template-tags (u/qualified-name tag) :dimension]))
 
 (mu/defn param-target->field-clause :- [:maybe mbql.s/field]
-  "Parse a Card parameter `target` form, which looks something like `[:dimension [:field-id 100]]`, and return the Field
-  ID it references (if any)."
+  "Parse a Card parameter `target` form, which looks something like `[:dimension [:field 100 nil]]`, and return the field
+  it references (if any)."
   [target card]
-  (let [target (mbql.normalize/normalize-tokens target :ignore-path)]
+  (let [target (mbql.normalize/normalize target)]
     (when (mbql.u/is-clause? :dimension target)
-      (let [[_ dimension] target]
-        (try
-          (unwrap-field-clause
-           (if (mbql.u/is-clause? :template-tag dimension)
-             (template-tag->field-form dimension card)
-             dimension))
-          (catch Throwable e
-            (log/error e (tru "Could not find matching Field ID for target:") target)))))))
+      (let [[_ dimension] target
+            field-form    (if (mbql.u/is-clause? :template-tag dimension)
+                            (template-tag->field-form dimension card)
+                            dimension)]
+        (when (or (integer? field-form)
+                  (mbql.u/is-clause? :field field-form))
```

Comment:
I'm confused here. What else can field-form be other than an integer, a field clause, or `nil`? If it can't be anything else, we should explicitly handle `nil`  here rather than have to list all the other types field-form could be.

Reply:
`field-form` can be an expression as well, but yea the check should be written as "not expression" rather than checking it's of certain types.

I've updated the log, please help taking another look.